projects/igniteui-angular/src/lib/grids/moving/moving.drag.directive.ts

```diff
@@ -23,35 +23,35 @@ export class IgxColumnMovingDragDirective extends IgxDragDirective implements On
         return this.cms.icon;
     }
 
-    protected _data: any;
     private subscription$: Subscription;
     private _ghostClass = 'igx-grid__drag-ghost-image';
     private ghostImgIconClass = 'igx-grid__drag-ghost-image-icon';
     private ghostImgIconGroupClass = 'igx-grid__drag-ghost-image-icon-group';
     private columnSelectedClass = 'igx-grid-th--selected';
 
     constructor(
-        public element: ElementRef<HTMLElement>,
-        public viewContainer: ViewContainerRef,
-        public zone: NgZone,
-        public renderer: Renderer2,
-        public cdr: ChangeDetectorRef,
+        element: ElementRef<HTMLElement>,
+        viewContainer: ViewContainerRef,
+        zone: NgZone,
+        renderer: Renderer2,
+        cdr: ChangeDetectorRef,
         private cms: IgxColumnMovingService,
         _platformUtil: PlatformUtil,
     ) {
         super(cdr, element, viewContainer, zone, renderer, _platformUtil);
     }
 
-    public ngOnDestroy() {
+    public override ngOnDestroy() {
         this._unsubscribe();
+        super.ngOnDestroy();
```

Comment:
@hanastasov There seems to be quite a lot of cleanup in the base `ngOnDestroy` so I added a call. Doesn't seem like those are overridden explicitly, but also behind some checks so not sure if they are in use in this directive and if the base call is really needed. Won't mind if someone takes a look,

Reply:
IIRC, same for the moving drop directive

Reply:
What's behind a check does not seem to be called even for the base directive. bcause the ghostElement is already cleared in `onTransitionEnd`. I believe the call is excessive, but does not do any harm